docs/powershell/customization.md

```diff
@@ -1,85 +1,454 @@
 
-The parameter `-Filter` is used to filter the results that a GET operation will return. 
-The parameter is defined as an String and follows a particular format of 
+### How are customizations used?
 
+Customizations are bundled together in their own module, `MyModule.custom`, which is referenced by the module that users ultimately download, `MyModule`. Running the `build-modules.ps1` script will create this new custom module and ensure that each customization is called in some fashion through the exported cmdlets.
+
+### Using the `PSBoundParameters` dictionary
+
+When a cmdlet is executed, the mapping of parameters provided and their corresponding values are stored in a variable called `PSBoundParameters`, which should be used to determine which parameters were provided by the user and can be passed along to any additional cmdlet calls.
+
+To determine if a parameter was provided by the user, the following check should be used:
+
+**PowerShell**
+
+```powershell
+if ($PSBoundParameters.ContainsKey("ParameterA"))
+{
+    # The variable $ParameterA has a value that was provided by the user
+}
+```
+
+The `PSBoundParameters` dictionary can also be passed to other cmdlets via splatting: rather than redefining all of the parameters and their values for the next cmdlet call, the cmdlet can use a special syntax to accept a dictionary, and all of the key-value pairs in the dictionary will be considered parameters and their corresponding values.
+
+To provide a dictionary, specifically `PSBoundParameters`, to another cmdlet via splatting, the following should be used:
+
+**PowerShell**
+
+```powershell
+MyModule\Get-Foo @PSBoundParameters
+```
+
+Before providing the `PSBoundParameters` dictionary to the next cmdlet, it should be verified that all of the parameters defined in this dictionary are available in the next cmdlet. If there is a parameter with a value in the dictionary, but it's not found in the next cmdlet, then it should be removed from the `PSBoundParameters` dictionary.
+
+To remove a parameter from the `PSBoundParameters` dictionary, the following should be used:
+
+**PowerShell**
+
+```powershell
+if ($PSBoundParameters.ContainsKey("Foo"))
+{
+    # The variable $Foo has a value that was provided by the user
+
+    # The -Foo parameter is not found in the next cmdlet, so we should remove it once we're done using it
+    $null = $PSBoundParameters.Remove("Foo")
+}
 ```
-eventTimestamp <op> <value> [and <attribute> eq <value> [and <attribute> eq <value>] ... ]
+
+_Note_: for PowerShell, the removal operation is assigned to a `$null` variable to prevent the returned `bool` from being written out from the cmdlet; this should be done for the removal operation on a dictionary.
```

Comment:
```suggestion
_Note_: for PowerShell, the removal operation is assigned to a `$null` variable to prevent the returned `bool` from being written to the Output stream from the cmdlet; this should be done for the removal operation on a dictionary, or any other command or call that returns a value.
```